### PR TITLE
Merge upstream changes with security fixes

### PR DESCRIPTION
**Pull request by unknwon**
https://github.com/russellhaering/gosaml2/security/advisories/GHSA-xhqq-x44f-9fgg

### CODE REVIEW

File: types/encrypted_key.go [hunk 1]
```diff
@@ -133,17 +159,31 @@ func (ek *EncryptedKey) DecryptSymmetricKey(cert *tls.Certificate) (cipher.Block
...
+			b, err := des.NewTripleDESCipher(pt)
 			if err != nil {
 				return nil, err
 			}
 
+			// FIXME: The version we had previously in our fork, AES seems more secure from my Googling.
+			// b, err := aes.NewCipher(pt)
+			// if err != nil {
+			// 	return nil, err
+			// }
```
Comment (unknwon) [inline types/encrypted_key.go:185]: @ElizabethStirling This is the only conflict that I can make a confident decision on, do you have an opinion?

FYI, Go tests passes either way.

Comment (ElizabethStirling) [reply types/encrypted_key.go:185]: AES sounds much better, let's go for that, if tests still pass with it.

Comment (unknwon) [reply types/encrypted_key.go:185]: Thanks for the input! I'm gonna merge as-is and address this in another commit so we don't lose track in this huge merge commit.

Comment (unknwon) [reply types/encrypted_key.go:185]: https://github.com/sourcegraph/gosaml2/commit/70944041979a86a57c73708d24d4b80b759d14ea

**Pull request merged**